Mathlib/MeasureTheory/Function/LpSpace.lean

```diff
@@ -672,6 +672,13 @@ theorem memℒp_indicator_iff_restrict (hs : MeasurableSet s) :
   simp [Memℒp, aestronglyMeasurable_indicator_iff hs, snorm_indicator_eq_snorm_restrict hs]
 #align measure_theory.mem_ℒp_indicator_iff_restrict MeasureTheory.memℒp_indicator_iff_restrict
 
+/-- For a function `f` with support in `s`, the Lᵖ norms of `f` with respect to `μ` and
+`μ.restrict s` are the same. -/
+theorem snorm_restrict_eq {s : Set α} (hs : MeasurableSet s) (hsf : f.support ⊆ s) :
```

Comment:
hs shouldn't be needed: otherwise, use the measurable hull of s instead. Since it contains s, it will also satisfy hsf

Reply:
I also thought that, but I wasn't sure how to get `snorm f p (μ.restrict s) = snorm f p (μ.restrict (toMeasurable μ s)`

Reply:
Loogle tells me about [MeasureTheory.Measure.restrict_toMeasurable_of_sFinite](https://leanprover-community.github.io/mathlib4_docs/Mathlib/MeasureTheory/Measure/Typeclasses.html#MeasureTheory.Measure.restrict_toMeasurable_of_sFinite)

Reply:
I replaced the lemma, and this is actually nicer. I didn't realize we had to put some effort in making the set measurable when we applied it, so this is a nice simplification.

This does add a small amount of imports to the file `LpSpace`.

Reply:
I have a proof removing the `SFinite` assumption, but I will first wait for #14260 to be merged, to avoid introducing new lemmas with a wrong name.

Reply:
See #14289